frontend/src/pages/projects/screens/spawner/storage/AddExistingStorageField.tsx

```diff
@@ -23,23 +26,65 @@ const AddExistingStorageField: React.FC<AddExistingStorageFieldProps> = ({
     currentProject,
     notebooks: { data: allNotebooks },
   } = React.useContext(ProjectDetailsContext);
-  const [pvcs, loaded, loadError] = useAvailablePvcs(
+  const [storages, loaded, loadError] = useAvailablePvcs(
     currentProject.metadata.name,
     allNotebooks,
     editStorage,
   );
+  const selectedStorage = data.storage;
+
+  const initialOptions = React.useMemo<TypeaheadSelectOption[]>(() => {
+    if (!loaded) {
+      return [];
+    }
+
+    const uniqueIdentifiers = new Set<PersistentVolumeClaimKind>();
+    storages.forEach((pvc) => {
+      uniqueIdentifiers.add(pvc);
+    });
+
+    return Array.from(uniqueIdentifiers).map((pvc) => ({
+      value: pvc.metadata.name,
+      content: getDisplayNameFromK8sResource(pvc),
+      selected: pvc.metadata.name === selectedStorage,
+    }));
+  }, [loaded, storages, selectedStorage]);
```

Comment:
Same question: Why was this change in logic necessary?

Reply:
fixed